lib/src/Uncertainty/Process/KarhunenLoeveQuadratureAlgorithm.cxx

```diff
@@ -0,0 +1,453 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief This class implements the computation of the Karhunen-Loeve
+ *         basis and eigenvalues of a given covariance model based on
+ *         quadrature approximation.
+ *
+ *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/KarhunenLoeveQuadratureAlgorithm.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/SquareMatrix.hxx"
+#include "openturns/SquareComplexMatrix.hxx"
+#include "openturns/NumericalSample.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/LinearNumericalMathFunction.hxx"
+#include "openturns/Pointer.hxx"
+#include "openturns/Uniform.hxx"
+#include "openturns/ComposedDistribution.hxx"
+#include "openturns/GaussProductExperiment.hxx"
+#include "openturns/EnumerateFunction.hxx"
+#include "openturns/OrthogonalProductPolynomialFactory.hxx"
+#include "openturns/LegendreFactory.hxx"
+#include "openturns/SpecFunc.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class KarhunenLoeveQuadratureAlgorithm
+ */
+
+CLASSNAMEINIT(KarhunenLoeveQuadratureAlgorithm);
+
+static const Factory<KarhunenLoeveQuadratureAlgorithm> Algorithm_KarhunenLoeveQuadratureAlgorithm;
+
+/* Constructor without parameters */
+KarhunenLoeveQuadratureAlgorithm::KarhunenLoeveQuadratureAlgorithm()
+  : KarhunenLoeveAlgorithmImplementation()
+  , domain_()
+  , basis_()
+  , basisSize_(0)
+  , experiment_()
```

Comment:
KarhunenLoeveQuadratureAlgorithm.cxx:54:5: warning: field 'basisSize_' will be initialized after field 'experiment_' [-Wreorder]